### ISSUE TITLE
Volumes Mount as Empty on Windows 19043 when Mounting Folders from Host over SMB

### ISSUE THREAD

**Issue by GuyPaddock** [docker/for-win#11427]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 22D5D8F0-AD23-4E4F-B320-38F9D8ED04F0/20210610201710

### Actual behavior
After upgrading Windows to 19043, it no longer appears possible to launch a Docker container from inside WSL2 using a folder from a mounted SMB/CIFS2 as a volume inside the container (i.e. if a folder in Windows is `C:\MyFolder`, and I have mounted the C drive over SMB as `/mnt/c-cifs`, then I am trying to start a container inside WSL2 that has a volume mounted from `/mnt/c-cifs/MyFolder`).

- After the upgrade, when running Docker for Windows 3.3.3, the volumes would mount as empty inside the containers.
- After upgrading to Docker for Windows 3.4.0, I now get this error when trying to start a container that uses a folder from a mounted SMB/CIFS as a volume inside the container:
```
ERROR: for arkayo_appserver_1  Cannot create container for service appserver: not a directory
ERROR: for appserver  Cannot create container for service appserver: not a directory
```
- If I launch the exact same container but reference the same folder via the drvfs mount (e.g. `/mnt/c/MyFolder`), the container starts but runs slowly (as documented for drvfs in WSL2 in https://github.com/microsoft/WSL/issues/4197).

### Expected behavior
Previously, Docker for Windows was able to mount folders that live on SMB mounts inside WSL2 inside containers without issue.

### Information
- Issue is 100% reproducible.
- Issue only started after an upgrade last night to Windows to 19043.

  - Windows Version: 10.0.19043 Build 19043
  - Docker Desktop Version: 3.4.0 (3.3.3 exhibited different behavior but still didn't function correctly, as described above).
  - WSL2 or Hyper-V backend?  WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Steps to reproduce the behavior
1. Install Docker for Windows.
2. Following steps in https://www.liip.ch/en/blog/setup-drupal-and-lando-with-wsl2-on-windows, install Lando inside Ubuntu 20.04.2 LTS on WSL2 using Deb packages but manually remove the hard dependency on `docker-ce` so that it uses the version of Docker provided by Docker for Windows instead.
3. Create a place for the `C` drive of the host system to be mounted (e.g. `sudo mkdir /mnt/c-cifs`).
4. Mount the `C` drive of the host system over SMB using a script like this (customize `USERNAME`):
```
HOST_IP=$(ipconfig.exe | grep WSL -A 4 | tail -n 1 | sed -E 's/.*: ([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})\r?\n?/\1/')
USERNAME="your.windows.username@example.com"

read -p "Windows password: " password

sudo mount -o username="${USERNAME}" -o password="${password}" -o uid=$(id -u) -o gid=$(id -g) -o vers=3 "//${HOST_IP}/c" /mnt/c-cifs/
```
5. Create the ["hello world" Lando project](https://docs.lando.dev/basics/first-app.html#hello-world) inside a folder mounted from the host over SMB (e.g. `/mnt/c-cifs/MyFolder`).
https://docs.lando.dev/basics/first-app.html#hello-world
6. Run `lando start` from within the hello-world project folder (e.g. `cd /mnt/c-cifs/MyFolder && lando start`)

The application will either fail to start with the error about "not a directory" or it will start but only serve-up "404 page not found".

**Comment by GuyPaddock** [comment]
Closing out -- this looks like a Lando issue. Here's why:
- The "Hello World" project always returns a 404 even when the folder is not mounted over SMB. (Ticketed this as https://github.com/lando/lando/issues/3038).
- The [Drupal 7 recipe](https://docs.lando.dev/basics/first-app.html#vanilla-drupal-7) for Lando appears to work fine over SMB, so this does not appear to be a Docker or WSL issue.

**Comment by GuyPaddock** [comment]
For others running into this issue:
- A full system reboot (not just a `lando rebuild`) resolved the `Cannot create container for service appserver: not a directory` message. I suspect restarting Docker Desktop might have resolved this as well; I remember having to do this a lot for this kind of error a few years back but had forgotten that sometimes this can happen since it hasn't happened in a long time.
- The empty mount problem was resolved by using "Reset to factory defaults" inside Docker (under Troubleshooting).

I reached the conclusion that resetting Docker might fix the problem when I observed that the Drupal 7 Lando recipe was mounting the SMB-hosted folder as a volume via a named mount like the following:
```
/run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu/1644e8b45b13530a823cabcef28cf8bc95190569e8f80b73be6763e0c4e4d5c4:/app:cached
```

While my other Lando app was mounting it as:
```
/mnt/c-cifs/MyFolder
```

So I concluded that perhaps Docker had something cached or conflicting that was preventing it from translating that path to a bind mount internally.
